src/js/actions/tools.js

```diff
@@ -419,9 +431,131 @@ define(function (require, exports) {
+        }
+
+        var toolMode = toolLib.toolModes.SHAPE;
+
+        if (vectorMode) {
+            toolMode = toolLib.toolModes.PATH;
+        }
+        
+        if (currentTool.id === "rectangle" || currentTool.id === "ellipse" || currentTool.id === "pen") {
+            var setObj = toolLib.setShapeToolMode(toolMode),
+                resetPromise = descriptor.batchPlayObjects([setObj]);
+
+            
+            if (!vectorMode && firstLaunch) {
+                var fillColor = [217, 217, 217],
+                    strokeColor = [157, 157, 157];
+
+                var defaultPromise = this.transfer(installShapeDefaults,
+                    currentTool.nativeToolName(), strokeColor, 2, 100, fillColor);
+
+                firstLaunch = false;
+                return Promise.join(initPromise, defaultPromise, resetPromise, dispatchPromise);
+            } else if (!vectorMode) {
+                return Promise.join(initPromise, resetPromise, dispatchPromise);
+            } else {
+                return Promise.join(initPromise, resetPromise, dispatchPromise)
+                .then(function () {
+                    return UI.setSuppressTargetPaths(false);
+                })
+                .then(function () {
+                    var pointerPolicy = new PointerEventPolicy(UI.policyAction.ALWAYS_PROPAGATE,
+                        OS.eventKind.LEFT_MOUSE_DOWN);
+                    return this.transfer(policy.addPointerPolicies, pointerPolicy);
+                })
+                .then(function () {
+                    return descriptor.playObject(vectorMaskLib.activateVectorMaskEditing());
+                });
+            }
+        } else if (currentTool.id === "newSelect" || currentTool.id === "superselectVector") {
+            if (!vectorMode) {
+                return this.transfer(selectTool, toolStore.getToolByID("newSelect"))
+                .then(function () {
+                    return Promise.join(initPromise, dispatchPromise);
+                });
+            } else {
+                return this.transfer(selectTool, toolStore.getToolByID("superselectVector"))
+                .then(function () {
+                    return Promise.join(initPromise, dispatchPromise);
+                })
+                .then(function () {
+                    return descriptor.playObject(vectorMaskLib.activateVectorMaskEditing());
+                })
+                .then(function () {
+                    descriptor.batchPlayObjects([vectorMaskLib.enterFreeTransformPathMode()],
+                        { synchronous: false });
+                })
+                .then(function () {
+                    var pointerPolicy = new PointerEventPolicy(UI.policyAction.ALWAYS_PROPAGATE,
+                        OS.eventKind.LEFT_MOUSE_DOWN);
+                    return this.transfer(policy.addPointerPolicies, pointerPolicy);
+                })
+                .catch(function () {
+                    // Silence the errors here
```

Comment:
What errors are we silencing here? What causes them to happen? Do we want to silence them, is this a quirk in PS logic? Is it possible that we'll ever get an error here we don't want to silence? Please be more explicit about this